file.go

```diff
@@ -0,0 +1,149 @@
+package gluster
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/gluster/gogfapi/gfapi"
+	billy "gopkg.in/src-d/go-billy.v4"
+)
+
+var (
+	_ billy.File = new(File)
+)
+
+type mode int
+
+const (
+	// ErrReadOnly returned when file is read only.
+	ErrReadOnly = "cannot write in %s, the file is read only"
+	// ErrWriteOnly returned when file is write only.
+	ErrWriteOnly = "cannot read from %s, the file is write only"
+
+	read  mode = 0
+	write mode = 1
+)
+
+// File holds a gluster file descritor.
+type File struct {
+	path  string
+	g     *gfapi.File
+	flags int
+}
+
+// NewFile creates a new wrapped gluster file descriptor.
+func NewFile(name string, g *gfapi.File, flags int) *File {
+	return &File{path: name, g: g, flags: flags}
+}
+
+// Name implements billy.File interface.
+func (f *File) Name() string {
+	return f.path
+}
+
+// Write implements billy.File interface.
+func (f *File) Write(p []byte) (int, error) {
+	err := f.checkFlags(write)
+	if err != nil {
+		return 0, err
+	}
+
+	return f.g.Write(p)
+}
+
+// Read implements billy.File interface.
+func (f *File) Read(p []byte) (int, error) {
+	err := f.checkFlags(read)
+	if err != nil {
+		return 0, err
+	}
+
+	n, err := f.g.Read(p)
+	// on error n is negative
+	if n < 0 {
+		n = 0
+	}
+	// it does not tell when the file ended, if we could not read the whole
+	// buffer treat it as EOF
+	if err == nil && n < len(p) {
+		err = io.EOF
```

Comment:
I wonder if it's ok to return `n>0` and `EOF`. I think people generally read till they get sth. 
Maybe we should return `n, nil` first and in next iteration `0, EOF` - WDYT?

Reply:
The tests for `go-billy` expect this behavior:

https://github.com/src-d/go-billy/blob/master/test/basic.go#L494-L498
```go
	b := make([]byte, 5)
	n, err := f.ReadAt(b, 0)
	c.Assert(err, Equals, io.EOF)
	c.Assert(n, Equals, 4)
	c.Assert(string(b), Equals, "TEST\x00")
```

What you describe is what made sense to me at first but seems that golang's file read behaves like that.